test/appium/tests/atomic/chats/test_public.py

```diff
@@ -437,23 +437,73 @@ def test_different_status_in_timeline(self):
+        message_2 = "testing no gap"
+        pub_chat_name = home_1.get_random_chat_name()
+        group_chat_name = home_1.get_random_chat_name()
+        public_key_1, username_1 = home_1.get_public_key_and_username(True)
+        public_key_2, username_2 = home_2.get_public_key_and_username(True)
         profile_1 = home_1.profile_button.click()
         profile_1.sync_settings_button.click()
         profile_1.sync_history_for_button.click()
         profile_1.element_by_translation_id("two-minutes").click()
         home_1.home_button.click()
-        chat_1, chat_2 = home_1.join_public_chat(chat_name), home_2.join_public_chat(chat_name)
-        chat_1.send_message("HI")
+        home_2.home_button.click()
+
+        home_1.just_fyi("Creating 1-1 chat and sending message from device 1")
+        one_to_one_chat_1 = home_1.add_contact(public_key_2)
+        one_to_one_chat_1.send_message("HI")
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Creating group chat and sending message from device 1")
+        group_chat_1 = home_1.create_group_chat([username_2], group_chat_name)
+        group_chat_1.send_message("HI")
+        home_1.get_back_to_home_view()
+
+        home_1.just_fyi("Creating public chat and sending message from device 1")
+        pub_chat_1, pub_chat_2 = home_1.join_public_chat(pub_chat_name), home_2.join_public_chat(pub_chat_name)
+        pub_chat_1.send_message("HI")
         device_1.toggle_airplane_mode()
-        chat_2.send_message(message)
+
+        home_2.just_fyi("Joining public chat by device 2 and sending message")
+        pub_chat_2.send_message(message_1)
+        home_2.get_back_to_home_view()
+
+        home_2.just_fyi("Joining 1-1 chat by device 2 and sending message")
+        one_to_one_chat_element = home_2.get_chat(username_1, wait_time=10)
+        one_to_one_chat_2 = one_to_one_chat_element.click()
+        one_to_one_chat_2.send_message(message_2)
+        home_2.get_back_to_home_view()
+
+        home_2.just_fyi("Joining Group chat by device 2 and sending message")
+        home_2.notifications_button.click()
+        group_chat_2 = home_2.get_chat_from_activity_center_view(group_chat_name).click()
+        group_chat_2.join_chat_button.click()
+        group_chat_2.send_message(message_2)
+
+        # Waiting for 3 minutes and then going back online
         sleep(180)
         device_1.toggle_airplane_mode()
-        chat_1.element_by_translation_id("fetch-messages").wait_and_click(60)
-        if not chat_1.chat_element_by_text(message).is_element_displayed():
+
+        home_1.just_fyi("Checking gap in public chat and fetching messages")
+        if pub_chat_1.chat_element_by_text(message_1).is_element_displayed(10):
+            device_1.driver.fail("Test message has been fetched automatically")
+        pub_chat_1.element_by_translation_id("fetch-messages").wait_and_click(60)
+        if not pub_chat_1.chat_element_by_text(message_1).is_element_displayed():
```

Comment:
btw is_element_displayed by default has 5 sec; in such types of tests will be better to add more time IMO or to wait implicitly when an app gets back to the online state, otherwise test may be flaky